flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosResourceManager.java

```diff
@@ -237,6 +244,10 @@ protected void initialize() throws ResourceManagerException {
 		Protos.FrameworkInfo.Builder frameworkInfo = mesosConfig.frameworkInfo()
 			.clone()
 			.setCheckpoint(true);
+		if (webUiUrl != null) {
+			frameworkInfo = frameworkInfo.setWebuiUrl(webUiUrl);
```

Comment:
It is not needed to reassign `frameworkInfo` because the method has side-effects on the builder. The return type is only a builder to enable method chaining. In fact, `frameworkInfo` can be declared `final`. 
```
if (webUiUrl != null) {
    frameworkInfo.setWebuiUrl(webUiUrl);
}
```